firebase-firestore/src/roboUtil/java/com/google/firebase/firestore/Values.java

```diff
@@ -22,62 +22,71 @@
 import com.google.protobuf.NullValue;
 import com.google.type.LatLng;
 import java.util.List;
+import java.util.Map;
 
 /** Test helper to create Firestore Value protos from Java types. */
-public class ValueHelper {
+public class Values {
 
+  // TODO(mrschmidt): Move into UserDataConverter
   public static Value valueOf(Object o) {
     if (o instanceof Value) {
       return (Value) o;
-    } else if (o instanceof String) {
-      return (Value.newBuilder().setStringValue((String) o).build());
+    } else if (o == null) {
+      return Value.newBuilder().setNullValue(NullValue.NULL_VALUE).build();
+    } else if (o instanceof Boolean) {
+      return Value.newBuilder().setBooleanValue((Boolean) o).build();
     } else if (o instanceof Integer) {
-      return (Value.newBuilder().setIntegerValue((long) (Integer) o).build());
+      return Value.newBuilder().setIntegerValue((Integer) o).build();
     } else if (o instanceof Long) {
-      return (Value.newBuilder().setIntegerValue((Long) o).build());
+      return Value.newBuilder().setIntegerValue((Long) o).build();
     } else if (o instanceof Double) {
-      return (Value.newBuilder().setDoubleValue((Double) o).build());
-    } else if (o instanceof Boolean) {
-      return (Value.newBuilder().setBooleanValue((Boolean) o).build());
+      return Value.newBuilder().setDoubleValue((Double) o).build();
     } else if (o instanceof Timestamp) {
       Timestamp timestamp = (Timestamp) o;
-      return (Value.newBuilder()
+      return Value.newBuilder()
           .setTimestampValue(
               com.google.protobuf.Timestamp.newBuilder()
                   .setSeconds(timestamp.getSeconds())
-                  .setNanos(timestamp.getNanoseconds())
-                  .build())
-          .build());
+                  .setNanos(timestamp.getNanoseconds()))
+          .build();
+    } else if (o instanceof String) {
+      return Value.newBuilder().setStringValue((String) o).build();
+    } else if (o instanceof Blob) {
+      return Value.newBuilder().setBytesValue(((Blob) o).toByteString()).build();
+
```

Comment:
nit: excess newline.

Reply:
Sorry - I forgot to address this before merging. I fixed it in the follow up PR.